webapp-django/crashstats/crashstats/static/crashstats/js/socorro/topcrash.js

```diff
@@ -45,25 +47,25 @@ var Correlations = (function() {
             platform: osname,
             signature: signature,
             product: SocReport.product,
-            version: SocReport.version
+            version: SocReport.version,
+            correlation_report_types: correlations,
         };
-        var types_done = 0;
-        $.each(correlations, function(i, type) {
-            data.correlation_report_type = type;
-            $.getJSON(SocReport.base, data, function(json) {
-                var report = '<h3>' + json.reason + '</h3>';
-                report += json.load.split("\n").join("<br>");
+
+        $.getJSON(SocReport.base + '?' + $.param(data, true))
+        .done(function(json) {
+            for (var i in correlations) {
+                var type = correlations[i];
+                var report = '<h3>' + json[type].reason + '</h3>';
+                report += json[type].load.split("\n").join("<br>");
                 row.find('.' + type).html(report);
-                types_done += 1;
+            }
```

Comment:
This ugly construction of a big fat blob of HTML as a string should die. But it is what it is. Hopefully we'll do it differently when we do Correlations on the Signature report.